tests/test_build.py

```diff
@@ -301,6 +301,46 @@ def test_sticky_header(sphinx_build_factory):
     assert index_html.select_one("body > .bd-header")
 
 
+def test_local_announcement_banner(sphinx_build_factory) -> None:
+    """If announcement is not a URL, it should be rendered at build time."""
+    confoverrides = {
+        "html_theme_options.announcement": "Hello, world!",
+    }
+    sphinx_build = sphinx_build_factory("base", confoverrides=confoverrides).build()
+    index_html = sphinx_build.html_tree("index.html")
+    results = index_html.find_all(class_="bd-header-announcement")
+
+    # Template should only render one announcement banner
+    assert len(results) == 1
+    banner = results[0]
+
+    # Announcement banner should contain the value from the config
+    assert banner.text.strip() == "Hello, world!"
+
+
+def test_remote_announcement_banner(sphinx_build_factory) -> None:
```

Comment:
These two tests are actually just unit tests for the `announcement.html` template. We don't really need to go through the sphinx_build_factory; I just don't know how to write this test in another way.